app/src/main/java/com/fstyle/structure_android/widget/dialog/DialogManagerImpl.java

```diff
@@ -1,51 +1,151 @@
 package com.fstyle.structure_android.widget.dialog;
 
-import android.app.Activity;
-import android.content.DialogInterface;
+import android.content.Context;
+import android.support.annotation.ArrayRes;
+import android.support.annotation.DrawableRes;
+import android.support.annotation.NonNull;
+import com.fstyle.library.DialogAction;
+import com.fstyle.library.MaterialDialog;
+import com.fstyle.structure_android.R;
 
 /**
  * Created by le.quang.dao on 14/03/2017.
  */
 
 public class DialogManagerImpl implements DialogManager {
 
-    private Activity mActivity;
-    private MainAlertDialog mDialog;
+    private static final String TAG = "DialogManagerImpl";
 
-    public DialogManagerImpl(Activity activity) {
-        mActivity = activity;
-    }
+    private Context mContext;
+    private MaterialDialog mProgressDialog;
 
-    @Override
-    public MainAlertDialog getDialog() {
-        return mDialog;
+    public DialogManagerImpl(Context context) {
+        mContext = context;
     }
 
     @Override
-    public boolean isShowing() {
-        if (mDialog == null) {
-            return false;
+    public void showIndeterminateProgressDialog() {
+        if (mProgressDialog == null) {
+            mProgressDialog = new MaterialDialog.Builder(mContext).content("Please wait…")
+                    .progress(true, 0)
+                    .build();
         }
-        return mDialog.isShowing();
+        mProgressDialog.show();
     }
 
     @Override
-    public void dismiss() {
-        if (mDialog == null) {
+    public void dismissProgressDialog() {
+        if (mProgressDialog == null) {
             return;
         }
-        mDialog.dismiss();
+        mProgressDialog.dismiss();
     }
 
     @Override
-    public void dialogMainStyle(String message,
-            DialogInterface.OnClickListener positiveButtonListener) {
-        if (isShowing()) {
-            return;
-        }
-        mDialog = new MainAlertDialog.Builder(mActivity).setVisibleTitle(false)
-                .setMessage(message)
-                .setPositiveListener(positiveButtonListener)
+    public void dialogBasicNoTitle(String content,
+            MaterialDialog.SingleButtonCallback positiveButtonListener) {
+        new MaterialDialog.Builder(mContext).content(content)
+                .positiveText("Agree")
```

Comment:
WARNING: [AvoidDuplicateLiterals] The String literal "Agree" appears 4 times in this file; the first occurrence is on line 48
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals